src/typegen.ts

```diff
@@ -26,12 +27,19 @@ export function doGenerate(
   sync: boolean,
   options: Options,
 ): void | Promise<void> {
+  const paginationStrategy = options.paginationStrategy
+  const prismaClientImportId =
+    Path.isAbsolute(options.typegenPath) &&
+    Path.isAbsolute(options.prismaClientPath)
+      ? Path.relative(options.typegenPath, options.prismaClientPath)
```

Comment:
This is the crux of the solution.

I'm conservative to only apply the relative solution when both inputs are absolute.